SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UnresolvedReferenceGymTest/UnresolvedReferenceGymTest.cpp

```diff
@@ -0,0 +1,57 @@
+
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "UnresolvedReferenceGymTest.h"
+
+AUnresolvedReferenceGymTest::AUnresolvedReferenceGymTest()
+	: Super()
+{
+	Author = "Ajanth";
+	Description = TEXT("Tests what happens when structs with references to actors whose entity have not been created yet are replicated.");
+}
+
+void AUnresolvedReferenceGymTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("MakeActorReferences"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AUnresolvedReferenceGymTestActor* ActorRefsActor =
+			GetWorld()->SpawnActor<AUnresolvedReferenceGymTestActor>({ 0.0f, 0.0f, 0.0f }, FRotator::ZeroRotator);
+		RegisterAutoDestroyActor(ActorRefsActor);
+
+		for (int i = 0; i < 3; i++)
+		{
+			AReplicatedTestActorBase* Actor =
+				GetWorld()->SpawnActor<AReplicatedTestActorBase>(FVector(0, 0, 0), FRotator::ZeroRotator, FActorSpawnParameters());
+			(ActorRefsActor->ActorRefs).Add(Actor);
+			RegisterAutoDestroyActor(Actor);
+		}
+
+		FinishStep();
+	});
+
+	AddStep(TEXT("CheckNoNullReferences"), FWorkerDefinition::AllClients, nullptr, nullptr, [this](float deltaTime) {
```

Comment:
I think what we might need to add is a step to wait for the AUnresolvedReferenceGymTestActor to be replicated to the client. Then we might want to add actors to it's array. @ImprobableNic will be able to talk more to how we need to set up the scenario to test correctly.

Reply:
Yep, I agree that would help the test being relevant.

Reply:
@ImprobableNic Do you know what the best way of reproducing unresolved references? Is there anything special we are missing? The previous manual test gym seemed to no actually produce any unresolved references. The array would contain all valid actors on first replication.